cuqi/model/_model.py

```diff
@@ -148,57 +148,82 @@ def domain_dim(self):
     def range_dim(self): 
         return self.range_geometry.par_dim
 
-    def _input2fun(self, x, geometry, is_par):
-        """ Converts input to function values (if needed) using the appropriate geometry. The input can then be passed to :class:`~cuqi.model.Model` operators (e.g. _forward_func, _adjoint_func, _gradient_func).
+    def _2fun(self, x, geometry, is_par):
+        """ Converts `x` to function values (if needed) using the appropriate 
+        geometry. For example, `x` can be the model input which need to be
+        converted to function value before being passed to 
+        :class:`~cuqi.model.Model` operators (e.g. _forward_func, _adjoint_func,
+        _gradient_func).
 
         Parameters
         ----------
         x : ndarray or cuqi.array.CUQIarray
-            The input value to be converted.
+            The value to be converted.
 
         geometry : cuqi.geometry.Geometry
-            The geometry representing the input `x`.
+            The geometry representing `x`.
 
         is_par : bool
-            If True the input is assumed to be parameters.
-            If False the input is assumed to be function values.
+            If True, `x` is assumed to be parameters.
+            If False, `x` is assumed to be function values.
 
         Returns
         -------
         ndarray or cuqi.array.CUQIarray
-            The input value represented as a function.
+            `x` represented as a function.
         """
-        if type(x) is CUQIarray and not isinstance(x.geometry, _DefaultGeometry1D):
-            return x.funvals
+        # Convert to function representation
+        # if x is CUQIarray and geometry are consistent, we obtain funvals
+        # directly
+        if isinstance(x, CUQIarray) and  x.geometry == geometry:
```

Comment:
Just to be sure. Can we compare the compute time when input is CUQIarray for Models now? Is the geometry check causing a lot of overhead?